code/jobs/DeployJob.php

```diff
@@ -132,23 +132,27 @@ public function perform() {
 				$this->args
 			);
 		} catch(Exception $e) {
+			// DeploynautJob will automatically trigger onFailure.
 			echo "[-] DeployJob failed" . PHP_EOL;
 			throw $e;
 		}
-		$this->updateStatus('Finished');
+		$this->updateStatus('complete');
 		echo "[-] DeployJob finished" . PHP_EOL;
 	}
 
+	public function onFailure(Exception $exception) {
+		$this->updateStatus('fail');
+	}
+
 	/**
-	 * @param string $status
+	 * @param string $status Transition
 	 * @global array $databaseConfig
 	 */
 	protected function updateStatus($status) {
```

Comment:
The latter - it's an API enforced by the DeploynautJob. And the other jobs are still using "Status" instead of state machines. Unfortunate, but forgivable I think